Remove THCTensor.cu and THCTensorCopy.cu copy

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65492
* __->__ #65491
* #65472
* #65471
* #65470

The only user of any of this code is THCStorage_copy, so I've
migrated that to call `Tensor.copy_` directly.

Differential Revision: [D31148183](https://our.internmc.facebook.com/intern/diff/D31148183)

peterbell10 committed:
Remove THCTensor.cu and THCTensorCopy.cu copy

The only user of any of this code is THCStorage_copy, so I've
migrated that to call `Tensor.copy_` directly.

[ghstack-poisoned]

peterbell10 committed:
Remove ATenCeilDiv in favour of ceil_div on "Remove THCTensor.cu and THCTensorCopy.cu copy"


The only user of any of this code is THCStorage_copy, so I've
migrated that to call `Tensor.copy_` directly.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THCTensor.cu and THCTensorCopy.cu copy"


The only user of any of this code is THCStorage_copy, so I've
migrated that to call `Tensor.copy_` directly.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THCTensor.cu and THCTensorCopy.cu copy"


The only user of any of this code is THCStorage_copy, so I've
migrated that to call `Tensor.copy_` directly.

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148183).

peterbell10 committed:
Remove manual hipifying include path on "Remove THCTensor.cu and THCTensorCopy.cu copy"


The only user of any of this code is THCStorage_copy, so I've
migrated that to call `Tensor.copy_` directly.

Differential Revision: [D31148183](https://our.internmc.facebook.com/intern/diff/D31148183)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148183).

ngimel commented:
Ugh, here apex is also failing with 
```
stderr: pytorch/apex/csrc/layer_norm_cuda_kernel.cu:4:10: fatal error: THC/THCDeviceUtils.cuh: No such file or directory
```
It doesn't appear to use anything from THCDeviceUtils.cuh, so just leaving empty file for now should be ok.

peterbell10 committed:
Update on "Remove THCTensor.cu and THCTensorCopy.cu copy"


The only user of any of this code is THCStorage_copy, so I've
migrated that to call `Tensor.copy_` directly.

Differential Revision: [D31148183](https://our.internmc.facebook.com/intern/diff/D31148183)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THCTensor.cu and THCTensorCopy.cu copy"


The only user of any of this code is THCStorage_copy, so I've
migrated that to call `Tensor.copy_` directly.

Differential Revision: [D31148183](https://our.internmc.facebook.com/intern/diff/D31148183)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THCTensor.cu and THCTensorCopy.cu copy"


The only user of any of this code is THCStorage_copy, so I've
migrated that to call `Tensor.copy_` directly.

Differential Revision: [D31148183](https://our.internmc.facebook.com/intern/diff/D31148183)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148183).
